### PR TITLE
Update staging-cd route53 records

### PR DESCRIPTION
**Pull request by aloftus23**
#  #

## 🗣 Description ##

Crossfeed's staging-cd infrastructure is no longer using Cloudfront, and has been replaced with API gateways. Therefore, the Route 53 record need to be updated to point at the APIs. 

## 💭 Motivation and context ##

Crossfeed's staging-cd infrastructure needs to match what we have in AWS GovCloud (doesn't support CloudFront).

## 🧪 Testing ##

- [x] Apply and test changes before merge

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] *All* future TODOs are captured in issues, which are referenced
      in code comments.
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated to reflect the changes in this PR.
- [x] All new and existing tests pass.

### CODE REVIEW

File: variables.tf
Comment (jsf9k) [inline variables.tf:1]: The changes in this file should definitely lead to changes in the `terraform-docs` output.

**Review by jsf9k — CHANGES_REQUESTED** [review]
I'm requesting that the `terraform-docs` output in `README.md` be updated.

@aloftus23 - Do you want me to go ahead and apply these changes to production so you can test that they function as expected?  We usually do that with these DNS record change PS.

**Comment by jsf9k** [comment]
@aloftus23 - Why did you remove the checklists from the PR template?

File: variables.tf
Comment (aloftus23) [reply variables.tf:1]: Ok I'll go ahead and make sure the README is updated. And yes that would be great if you can apply the changes for testing.

**Comment by aloftus23** [comment]
> @aloftus23 - Why did you remove the checklists from the PR template?

I copied over a previous PR. I'll add them back.

File: variables.tf [hunk 1]
```diff
@@ -45,9 +45,15 @@ variable "aws_region" {
...
+variable "api_gateway_east_1_zone_id" {
   type        = string
-  description = "The ID of the Cloudfront hosted zone. This is set by AWS and is a constant across all Cloudfront distributions."
+  description = "The Route 53 hosted zone ID for For us-east-1 endpoints."
+  default     = "Z1UJRXOUMOOFQ8"
+}
+
+variable "api_gateway_edge_zone_id" {
+  type        = string
+  description = "The Route 53 hosted zone ID for For edge-optimized endpoints."
```
Comment (dav3r) [inline variables.tf]: Remove duplicated word:
```suggestion
  description = "The Route 53 hosted zone ID for us-east-1 endpoints."
  default     = "Z1UJRXOUMOOFQ8"
}

variable "api_gateway_edge_zone_id" {
  type        = string
  description = "The Route 53 hosted zone ID for edge-optimized endpoints."
```

Also, these variables should be moved so that they are correctly alphabetized within this file.

**Comment by jsf9k** [comment]
@aloftus23 - I get this when I try to apply:
```
Error: creating Route 53 Record: InvalidChangeBatch: [Tried to create an alias that targets mkrl99gnu9.execute-api.us-east-1.amazonaws.com., type A in zone Z2FDTNDATAQYW2, but the alias target name does not lie within the target zone]
│       status code: 400, request id: f43bb7d7-03f4-409e-940a-76831b2ce1c9
│ 
│   with aws_route53_record.crossfeed_staging_cd_api_A,
│   on route53_crossfeed_app.tf line 239, in resource "aws_route53_record" "crossfeed_staging_cd_api_A":│  239: resource "aws_route53_record" "crossfeed_staging_cd_api_A" {
│ 
╵
╷
│ Error: creating Route 53 Record: InvalidChangeBatch: [Tried to create an alias that targets mkrl99gnu9.execute-api.us-east-1.amazonaws.com., type AAAA in zone Z2FDTNDATAQYW2, but the alias target name does not lie within the target zone]
│       status code: 400, request id: 01e40ed0-9f35-4707-9b6d-97b5c39a1483
│ 
│   with aws_route53_record.crossfeed_staging_cd_api_AAAA,
│   on route53_crossfeed_app.tf line 252, in resource "aws_route53_record" "crossfeed_staging_cd_api_AAAA":
│  252: resource "aws_route53_record" "crossfeed_staging_cd_api_AAAA" {
```

File: variables.tf [hunk 2]
```diff
@@ -39,18 +39,24 @@ variable "additional_ses_sendemail_account_ids" {
   default     = []
 }
 
-variable "aws_region" {
+variable "api_gateway_east_1_zone_id" {
   type        = string
-  description = "The AWS region to communicate with."
-  default     = "us-east-1"
+  description = "The Route 53 hosted zone ID for us-east-1 endpoints."
+  default     = "Z1UJRXOUMOOFQ8"
```
Comment (dav3r) [inline variables.tf]: Since we already specify the region for this deployment in the `aws_region` variable, this variable should be renamed `api_gateway_zone_id` and the description should remove the reference to `us-east-1`.

File: variables.tf
Comment (jsf9k) [reply variables.tf:1]: @aloftus23 - I get this when I try to apply:
```
Error: creating Route 53 Record: InvalidChangeBatch: [Tried to create an alias that targets mkrl99gnu9.execute-api.us-east-1.amazonaws.com., type A in zone Z2FDTNDATAQYW2, but the alias target name does not lie within the target zone]
│       status code: 400, request id: f43bb7d7-03f4-409e-940a-76831b2ce1c9
│ 
│   with aws_route53_record.crossfeed_staging_cd_api_A,
│   on route53_crossfeed_app.tf line 239, in resource "aws_route53_record" "crossfeed_staging_cd_api_A":│  239: resource "aws_route53_record" "crossfeed_staging_cd_api_A" {
│ 
╵
╷
│ Error: creating Route 53 Record: InvalidChangeBatch: [Tried to create an alias that targets mkrl99gnu9.execute-api.us-east-1.amazonaws.com., type AAAA in zone Z2FDTNDATAQYW2, but the alias target name does not lie within the target zone]
│       status code: 400, request id: 01e40ed0-9f35-4707-9b6d-97b5c39a1483
│ 
│   with aws_route53_record.crossfeed_staging_cd_api_AAAA,
│   on route53_crossfeed_app.tf line 252, in resource "aws_route53_record" "crossfeed_staging_cd_api_AAAA":
│  252: resource "aws_route53_record" "crossfeed_staging_cd_api_AAAA" {
```

Comment (jsf9k) [reply variables.tf:1]: Maybe these should instead be `CNAME` records?  I'm not sure.

File: variables.tf (hunk 2 above)
Comment (aloftus23) [reply variables.tf]: Done.

File: README.md [hunk 3]
```diff
@@ -151,6 +151,8 @@ zone.  This role has a trust relationship with the users account.
 | acmresourcechange\_role\_name | The name to assign the IAM role (as well as the corresponding policy) that allows sufficient permissions to modify ACM (AWS Certificate Manager) resources in the DNS account. | `string` | `"ACMResourceChange"` | no |
 | additional\_remote\_state\_account\_ids | A list of account IDs corresponding to additional accounts that should have permission to assume the role to read this root module's remote state (e.g. ["123456789012"]). | `list(string)` | `[]` | no |
 | additional\_ses\_sendemail\_account\_ids | A list of account IDs corresponding to additional accounts that should have permission to assume the role to send email via AWS SES (e.g. ["123456789012"]). | `list(string)` | `[]` | no |
+| api\_gateway\_east\_1\_zone\_id | The Route 53 hosted zone ID for For us-east-1 endpoints. | `string` | `"Z1UJRXOUMOOFQ8"` | no |
+| api\_gateway\_edge\_zone\_id | The Route 53 hosted zone ID for For edge-optimized endpoints. | `string` | `"Z2FDTNDATAQYW2"` | no |
```
Comment (jsf9k) [inline README.md]:
```suggestion
| api\_gateway\_east\_1\_zone\_id | The Route 53 hosted zone ID for us-east-1 endpoints. | `string` | `"Z1UJRXOUMOOFQ8"` | no |
| api\_gateway\_edge\_zone\_id | The Route 53 hosted zone ID for edge-optimized endpoints. | `string` | `"Z2FDTNDATAQYW2"` | no |
```

**Review by jsf9k — CHANGES_REQUESTED** [review]
You have a previous typo that got propagated.

File: variables.tf
Comment (aloftus23) [reply variables.tf:1]: I think it has to be in the us-east-1 zone actually. Do you mind trying again now?

Comment (jsf9k) [reply variables.tf:1]: Yep, I was able to apply this time.  You can verify functionality now.

Comment (jsf9k) [reply variables.tf:1]: See commit 9d3eed1.

File: README.md (hunk 3 above)
Comment (jsf9k) [reply README.md]: See commit d723e73a878c25b9160e76f5c5fc5e71badd2a9b.

File: variables.tf [hunk 4]
```diff
@@ -39,16 +39,16 @@ variable "additional_ses_sendemail_account_ids" {
   default     = []
 }
 
-variable "aws_region" {
+variable "api_gateway_zone_id" {
   type        = string
-  description = "The AWS region to communicate with."
-  default     = "us-east-1"
+  description = "The Route 53 hosted zone ID for us-east-1 endpoints."
```
Comment (jsf9k) [inline variables.tf]: The description shouldn't mention `us-east-1`, as @dav3r pointed out [here](https://github.com/cisagov/cool-dns-cyber.dhs.gov/pull/92#discussion_r1481987693).

File: README.md [hunk 5]
```diff
@@ -151,8 +152,8 @@ zone.  This role has a trust relationship with the users account.
 | acmresourcechange\_role\_name | The name to assign the IAM role (as well as the corresponding policy) that allows sufficient permissions to modify ACM (AWS Certificate Manager) resources in the DNS account. | `string` | `"ACMResourceChange"` | no |
 | additional\_remote\_state\_account\_ids | A list of account IDs corresponding to additional accounts that should have permission to assume the role to read this root module's remote state (e.g. ["123456789012"]). | `list(string)` | `[]` | no |
 | additional\_ses\_sendemail\_account\_ids | A list of account IDs corresponding to additional accounts that should have permission to assume the role to send email via AWS SES (e.g. ["123456789012"]). | `list(string)` | `[]` | no |
+| api\_gateway\_zone\_id | The Route 53 hosted zone ID for us-east-1 endpoints. | `string` | `"Z1UJRXOUMOOFQ8"` | no |
```
Comment (jsf9k) [inline README.md]: See [here](https://github.com/cisagov/cool-dns-cyber.dhs.gov/pull/92#discussion_r1482007816).

**Review by jsf9k — CHANGES_REQUESTED** [review]

File: README.md [hunk 6]
```diff
@@ -106,6 +106,7 @@ zone.  This role has a trust relationship with the users account.
 | [aws_route53_record.root_MX](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/route53_record) | resource |
 | [aws_route53_record.root_SPF](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/route53_record) | resource |
 | [aws_route53_record.root_acm_rules_CNAME](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/route53_record) | resource |
+| [aws_route53_record.rsaa_dev_CNAME](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/route53_record) | resource |
```
Comment (dav3r) [inline README.md]: Why is this here?  It seems to be outside the scope of this PR.

Comment (aloftus23) [reply README.md]: Running the terraform-docs command added it. I'll look into why

**Comment by aloftus23** [comment]
@jsf9k Something didn't work after your apply. Debugging now and will let you know.

File: variables.tf [hunk 7]
```diff
@@ -39,16 +39,16 @@ variable "additional_ses_sendemail_account_ids" {
   default     = []
 }
 
-variable "aws_region" {
+variable "api_gateway_zone_id" {
   type        = string
-  description = "The AWS region to communicate with."
-  default     = "us-east-1"
+  description = "The Route 53 hosted zone ID for API Gateways."
```
Comment (dav3r) [inline variables.tf]: Remove unnecessary capitalization:
```suggestion
  description = "The Route 53 hosted zone ID for API gateways."
```

File: README.md [hunk 8]
```diff
@@ -151,8 +151,8 @@ zone.  This role has a trust relationship with the users account.
 | acmresourcechange\_role\_name | The name to assign the IAM role (as well as the corresponding policy) that allows sufficient permissions to modify ACM (AWS Certificate Manager) resources in the DNS account. | `string` | `"ACMResourceChange"` | no |
 | additional\_remote\_state\_account\_ids | A list of account IDs corresponding to additional accounts that should have permission to assume the role to read this root module's remote state (e.g. ["123456789012"]). | `list(string)` | `[]` | no |
 | additional\_ses\_sendemail\_account\_ids | A list of account IDs corresponding to additional accounts that should have permission to assume the role to send email via AWS SES (e.g. ["123456789012"]). | `list(string)` | `[]` | no |
+| api\_gateway\_zone\_id | The Route 53 hosted zone ID for API Gateways. | `string` | `"Z1UJRXOUMOOFQ8"` | no |
```
Comment (dav3r) [inline README.md]: Remove unnecessary capitalization:
```suggestion
| api\_gateway\_zone\_id | The Route 53 hosted zone ID for API gateways. | `string` | `"Z1UJRXOUMOOFQ8"` | no |
```

File: README.md (hunk 6 above)
Comment (jsf9k) [reply README.md]: `terraform-docs` may be looking at the remote state to construct its output. Those resources do exist in the state.  I put them out there so #93 can test their changes too.  They can simply be removed from `README.md` for the purposes of this PR.

File: variables.tf (hunk 4 above)
Comment (jsf9k) [reply variables.tf]: See commit b56552c9fc3074bb725b54bb94372b601cb00a0f.

File: README.md (hunk 5 above)
Comment (jsf9k) [reply README.md]: See commit b56552c9fc3074bb725b54bb94372b601cb00a0f.

File: variables.tf (hunk 7 above)
Comment (aloftus23) [reply variables.tf]: Fixed in this commit [c14aa64](https://github.com/cisagov/cool-dns-cyber.dhs.gov/pull/92/commits/c14aa64dca18d92c60eec7aab6b358b685090546)

File: README.md (hunk 8 above)
Comment (aloftus23) [reply README.md]: Fixed in this commit [c14aa64](https://github.com/cisagov/cool-dns-cyber.dhs.gov/pull/92/commits/c14aa64dca18d92c60eec7aab6b358b685090546)

**Comment by aloftus23** [comment]
@jsf9k Ok we are ready to test another apply when you get the chance.

Realized we were pointing at the API gateways with the wrong domain name.

File: variables.tf (hunk 7 above)
Comment (dav3r) [reply variables.tf]: Thanks, though in the future, please create separate commits for the capitalization fix and the changes to the alias domain.

**Comment by jsf9k** [comment]
> @jsf9k Ok we are ready to test another apply when you get the chance.
> 
> Realized we were pointing at the API gateways with the wrong domain name.

I reapplied.

File: README.md (hunk 6 above)
Comment (dav3r) [reply README.md]: Looks like this was removed in https://github.com/cisagov/cool-dns-cyber.dhs.gov/pull/92/commits/b56552c9fc3074bb725b54bb94372b601cb00a0f, so I am resolving this conversation.

**Comment by jsf9k** [comment]
@aloftus23 - I need to rebase this PR and push to your branch, but I don't have permission.  I think there is a box you can check on your PR that will allow this.  Can you check that box?

**Comment by aloftus23** [comment]
I'm only seeing this already checked. 
Screenshot 2024-02-08 at 11 29 17 AM

If we just need to update the fork with the base branch, I can go ahead and do that.

**Comment by aloftus23** [comment]
> I'm only seeing this already checked. Screenshot 2024-02-08 at 11 29 17 AM
> 
> If we just need to update the fork with the base branch, I can go ahead and do that.

Should be inline with develop now. Also, we are testing and everything is looking good so far. Will lyk when we are 100%

**Review by jsf9k — APPROVED** [review]
Thanks sticking with us through the approval process!

**Review by dav3r — APPROVED** [review]
👍

**Comment by aloftus23** [comment]
No problem, thank you all for the quick reviews! We've completed testing and confirmed everything is working. So good to merge when you are.
